server/channels/app/permissions_migrations.go

```diff
@@ -1092,6 +1092,31 @@ func (a *App) getProductsBoardsPermissions() (permissionsMap, error) {
 	return transformations, nil
 }
 
+func (a *App) getAddChannelReadContentPermissions() (permissionsMap, error) {
+	t := []permissionTransformation{}
+
+	readChannelContentPermissions := []string{
+		model.PermissionReadChannelContent.Id,
+	}
+
+	// Migrate all roles including custom roles that have the read_channel permission
+	// but exclude system console roles system_read_only_admin system_user_manager & system_manager
+	// as this system roles are for the admin console use only
+	t = append(t, permissionTransformation{
+		On: permissionAnd(
+			permissionAnd(
+				isNotExactRole(model.SystemUserManagerRoleId),
```

Comment:
There is a deprecation warning for this, did you check that?

Reply:
What deprecation?

I'm aware that this role is directly responsible for this PR and users can still get assign this role through the admin console.

You think it can represent a problem?

Reply:
The method is deprecated:
https://github.com/mattermost/mattermost-server/blob/a36072e0d80dd69b58598339726282288334a958/server/channels/app/permissions_migrations.go#L109

Reply:
Oh I see.. let me use the new method instead...